client/src/prediction.rs

```diff
@@ -13,21 +13,31 @@ pub struct PredictedMotion<T: Clone> {
     log: VecDeque<T>,
     generation: u16,
     predicted_position: Position,
+    predicted_velocity: na::Vector3<f32>,
 }
 
 impl<T: Clone> PredictedMotion<T> {
-    pub fn new(initial_position: Position) -> Self {
+    pub fn new(initial_position: Position, initial_velocity: na::Vector3<f32>) -> Self {
```

Comment:
Are we ever going to want to pass anything other than the zero vector for `initial_velocity`? If you can't immediately think of a case, just bake that in for simplicity for now; easy to change later if needed.